engine-ext/api/src/main/java/org/pentaho/di/engine/api/remote/StopMessage.java

```diff
@@ -25,50 +25,127 @@
 package org.pentaho.di.engine.api.remote;
 
 /**
- * Close Message
+ * Used for requesting to stop execution and for informing the sessionKilled of the stop operation.
+ * Also used to inform the client of server failure
  * <p>
  * Since the CloseReason is not serializable collect the necessary parts and reconstruct the object when needed.
  * <p>
  * NOTE: There was an attempt to extend CloseReason and implement Message to make it serializable; but this did not
  * work.
  * <p>
- * TODO Add CloseReason if it is ok to add references to the javax.websocket-api depencency Created by ccaspanello on
+ * Created by ccaspanello on
  * 7/25/17.
  */
 public class StopMessage implements Message {
 
+  public enum Status {
+    SESSION_KILLED, // spark session was killed
+    SUCCESS, //Stop execution was successful
+    FAILED //failed to stop execution
+  }
+
   private static final long serialVersionUID = 8842623444691045346L;
-  //private CloseReason.CloseCode closeCode;
   private String reasonPhrase;
   private String requestUUID;
+  private Status result;
 
-  public StopMessage(
-    //CloseReason.CloseCode closeCode,
-    String reasonPhrase ) {
-    //this.closeCode = closeCode;
+  /**
+   * Constructor used by the clients when sending the stop request for the daemon server
+   *
+   * @param reasonPhrase reason for the Stop request
+   */
+  public StopMessage( String reasonPhrase ) {
     this.reasonPhrase = reasonPhrase;
+    this.result = Status.SUCCESS;
```

Comment:
since the you're defaulting the result to success, any reason not to do this in the variable initializer?  so you don't have to do it in each constructor